k8sClient.py

```diff
@@ -96,3 +97,22 @@ def wait_for_crd(self, name: str, cr_name: str, namespace: str) -> None:
             retries -= 1
             if retries <= 0:
                 logger.error_and_exit(f"Failed to get cr {cr_name}/{name}")
+
+    def wait_for_all_pods_in_namespace(self, namespace: str, condition: str = "Ready") -> None:
+        logger.info(f"Waiting for all pods in namespace {namespace} to be '{condition}'")
+        it = 0
+        retries = 10
```

Comment:
Seems like 200 minutes of waiting is pretty excessive in the case pods are failing to reach the desired state. Maybe we can reduce this.

Also keep in mind there could be a race condition when this is called if the pods are not yet being created. I.e. SNO pod is creating and starts spinning up the daemon or webhook pods. This check is passed before these pods are installed, at which point these pods get added and end up in a crashloop backoff.

Maybe we need an additional check of the deployment itself, or a sleep to ensure things have time to settle.